### DISCUSSION TITLE
DPO

### DISCUSSION THREAD

**Discussion by fakerybakery** [axolotl-ai-cloud/axolotl#1008] [Q&A] [answered]
Hi @winglian,
Is it possible to train models with DPO (Direct Performance Optimization) using this framework?
Thank you!

**Comment by araleza** [comment]
I don't have the answer this this (sorry), but I just wanted to say that I'd also like to know this.  I've seen people using DPO to train their models and they say they get significant quality gains, but I've never seen anyone explain what the software or setup used to do this is.

**Reply by fakerybakery** [reply]
FYI here's the [DPO paper](https://arxiv.org/abs/2305.18290)

**Comment by filippo82** [comment, marked as answer]
Yes, it is possible but you currently need to checkout the [rl-trainer](https://github.com/OpenAccess-AI-Collective/axolotl/tree/rl-trainer) branch. You can find an example configuration [here](https://huggingface.co/openaccess-ai-collective/DPOpenHermes-7B-v2/tree/main/configs).

**Comment by NanoCode012** [comment]
Fyi, this has also been merged to main in beta! https://github.com/OpenAccess-AI-Collective/axolotl/blob/main/docs/rlhf.md
